From 685565f7603e4b2a19e8329dbea60e0d1059d9f3 Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Wed, 2 Aug 2017 21:32:55 +0200
Subject: [PATCH] cmd/snap-confine: don't share /etc/nsswitch from host

This patch corrects name resolution errors on Arch where host's
/etc/nsswitch.conf contains references to modules not present in the
core snap. This is again one of the bugs originating from sharing all of
/etc blindly and then undoing the damage but for now it is easier to add
an exception than to undo this.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 cmd/snap-confine/mount-support.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmd/snap-confine/mount-support.c b/cmd/snap-confine/mount-support.c
index bd028c7a5..c906459ce 100644
--- a/cmd/snap-confine/mount-support.c
+++ b/cmd/snap-confine/mount-support.c
@@ -325,7 +325,9 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		//  - https://bugs.launchpad.net/snap-confine/+bug/1580018
 		//  - https://bugzilla.opensuse.org/show_bug.cgi?id=1028568
 		const char *dirs_from_core[] =
-		    { "/etc/alternatives", "/etc/ssl", NULL };
+		    { "/etc/alternatives", "/etc/ssl", "/etc/nsswitch.conf",
+			NULL
+		};
 		for (const char **dirs = dirs_from_core; *dirs != NULL; dirs++) {
 			const char *dir = *dirs;
 			if (access(dir, F_OK) == 0) {
-- 
2.13.3

